### PR TITLE
#165437143 Account transaction history endpoint

### PR DESCRIPTION
**Pull request by Shegsdev**
### What does this PR do?
Re-configure database and
create account transaction history endpoint.

### Description of Task to be completed?
Have the following endpoint working
- GET /api/v1/accounts/:accountNumber/transactions

#### Previous endpoints (using database).
- POST /api/v1/auth/signup
- POST /api/v1/auth/signin
- GET /api/v1/accounts/:acccountNumber
- PATCH /api/v1/accounts/:acccountNumber
- POST /api/v1/accounts
- GET /api/v1/accounts
- POST /api/v1/transactions/:acccountNumber/credit
- POST /api/v1/transactions/:acccountNumber/debit

### What are the relevant pivotal tracker stories?
[#165437143](https://www.pivotaltracker.com/story/show/165437143/comments/201913106)

### CODE REVIEW

File: server/controllers/accounts.controller.js [hunk 1]
```diff
@@ -9,52 +12,76 @@ const AccountsController = {
...
+    // Generate account account number
+    const date = new Date();
+    const accountNumber = date.getTime();
+
+    // Check if user has a registered account
+    User.findOne('email', email)
+      .then((result) => {
+        if (!result || result.rows.length < 1) {
+          return res.status(400).json({
+            status: 400,
```
Comment (kampkelly) [inline server/controllers/accounts.controller.js]: Status code here is not correct. Please change it.

File: server/controllers/accounts.controller.js [hunk 2]
```diff
@@ -9,52 +12,76 @@ const AccountsController = {
...
+              firstName,
+              lastName,
+              email,
+              type: account.type,
+              openingBalance: account.balance,
+            },
+          }))
+          .catch(err => res.status(400).json({
+            status: 400,
+            error: `Could not create account ${err}`,
```
Comment (kampkelly) [inline server/controllers/accounts.controller.js]: Change status code here too.

File: server/controllers/accounts.controller.js [hunk 3]
```diff
@@ -70,27 +97,59 @@ const AccountsController = {
...
+          status: 404,
+          error: 'Acount does not exist',
+        });
+      }
     }
-    return res.status(200).json({
-      status: 200,
-      data: account,
-    });
+    // return res.status(200).json({ status: 200, data: '' });
```
Comment (kampkelly) [inline server/controllers/accounts.controller.js]: Remove commented line.

File: server/controllers/accounts.controller.js [hunk 4]
```diff
@@ -104,20 +163,27 @@ const AccountsController = {
...
+          status: 200,
+          data: {
+            accountNumber: data.accountNumber,
+            status: data.status,
+          },
+        });
+      })
+      .catch(err => res.status(400).json({
+        status: 400,
+        error: `Error updating account ${err}`,
```
Comment (kampkelly) [inline server/controllers/accounts.controller.js]: Please change status code.

File: server/controllers/auth/signin.controller.js [hunk 5]
```diff
@@ -3,7 +3,7 @@ import bcrypt from 'bcrypt';
 import jwt from 'jsonwebtoken';
 import { config } from 'dotenv';
 import User from '../../models/user.model';
-import { setAuthToken } from '../../utils/helpers';
+// import { setAuthToken } from '../../utils/helpers';
 import validateSignInInput from '../../validation/authentication/signin';
```
Comment (kampkelly) [inline server/controllers/auth/signin.controller.js]: Remove commented line.

File: server/controllers/auth/signin.controller.js [hunk 6]
```diff
@@ -35,53 +35,55 @@ const SigninController = {
...
     if (!isValid) {
-      return res.status(400).send({
-        status: 400,
-        error,
-      });
+      return res.status(400).send({ status: 400, error });
     }
 
+    // Convert email to lowercase
+    email = email.toLowerCase().trim();
```
Comment (kampkelly) [inline server/controllers/auth/signin.controller.js]: Remove this comment. The code is self-explanatory. Also, remove comments for any line of code that is self-explanatory.

File: server/controllers/auth/signin.controller.js [hunk 7]
```diff
@@ -35,53 +35,55 @@ const SigninController = {
...
+        // Check if password match
+        bcrypt.compare(password, result.rows[0].password)
+          .then((isMatch) => {
+            if (isMatch) {
+              const payload = result.rows[0];
+              jwt.sign(payload, secret, { expiresIn: '2h' }, (err, token) => {
+                if (err) {
+                  return res.status(500).json({
+                    status: 500,
+                    error: `Some error occured - ${err}`,
```
Comment (kampkelly) [inline server/controllers/auth/signin.controller.js]: Change status code.

File: server/controllers/auth/signup.controller.js [hunk 8]
```diff
@@ -36,68 +34,61 @@ const SignupController = {
...
+            });
           });
         });
-      })
-      .catch(err => res.status(500).json({
-        status: 500,
-        error: `Something went wrong ${err}. Please try again`,
-      }));
+      // .catch(_err => console.log('Something went wrong. Please try again'));
+    });
```
Comment (kampkelly) [inline server/controllers/auth/signup.controller.js]: Remove commented line.

File: server/controllers/transactions.controller.js [hunk 9]
```diff
@@ -17,68 +17,53 @@ const TransactionsController = {
...
+    let { amount } = req.body;
     amount = parseFloat(amount);
-    cashier = parseInt(cashier, 10);
     const { error, isValid } = validateTransactionInput(req.body);
     if (!isValid) {
-      return res.status(400).json({
-        status: 400,
-        error,
-      });
+      return res.status(400).json({ status: 400, error });
```
Comment (kampkelly) [inline server/controllers/transactions.controller.js]: Change status code.

File: server/controllers/transactions.controller.js [hunk 10]
```diff
@@ -17,68 +17,53 @@ const TransactionsController = {
...
+            amount: newTransaction.rows[0].amount,
+            cashier: newTransaction.rows[0].cashier,
+            transactionType: newTransaction.rows[0].type,
+            accountBalance: newTransaction.rows[0].new_balance,
+          },
+        });
+      } catch (err) {
+        res.status(400).json({
+          status: 400,
+          error: `There was an error making new transaction ${err}`,
```
Comment (kampkelly) [inline server/controllers/transactions.controller.js]: Change status code.

File: server/controllers/accounts.controller.js (hunk 1 above)
Comment (Shegsdev) [reply server/controllers/accounts.controller.js]: Changed to
#### Not Found.

File: server/controllers/accounts.controller.js (hunk 2 above)
Comment (Shegsdev) [reply server/controllers/accounts.controller.js]: I changed this to server error(500). Is it good?

File: server/controllers/accounts.controller.js (hunk 1 above)
Comment (kampkelly) [reply server/controllers/accounts.controller.js]: I do not understand what you mean by Not found. I cannot see it in the code.

File: server/controllers/accounts.controller.js (hunk 2 above)
Comment (kampkelly) [reply server/controllers/accounts.controller.js]: Please where is the change? I can't see it.

File: server/controllers/accounts.controller.js [hunk 11]
```diff
@@ -9,52 +12,76 @@ const AccountsController = {
...
    *
    * */
   async create(req, res) {
     const { error, isValid } = validateCreateBankAccountInput(req.body);
     if (!isValid) {
-      return res.status(400).json({
-        status: 400,
-        error,
-      });
+      return res.status(406).json({ status: 406, error });
```
Comment (kampkelly) [inline server/controllers/accounts.controller.js]: Please change this from 406.

File: server/controllers/accounts.controller.js [hunk 12]
```diff
@@ -70,27 +97,57 @@ const AccountsController = {
...
+
+      if (!accountNumber || accountNumber.toString().length < 13) {
+        return res.status(401).json({ status: 401, error: 'Invalid account number' });
+      }
+      return res.status(200).json({ status: 200, data: transactions.rows });
+    } catch (err) {
+      if (err) {
+        return res.status(404).json({
+          status: 404,
+          error: 'Acount does not exist',
```
Comment (kampkelly) [inline server/controllers/accounts.controller.js]: Please change this status code.

File: server/controllers/accounts.controller.js [hunk 13]
```diff
@@ -127,24 +191,24 @@ const AccountsController = {
...
    * Route: DELETE: /accounts/:accountNumber
    *
    * */
   delete(req, res) {
     if (!req.params.accountNumber) {
-      res.status(400).json({
-        status: 400,
+      res.status(406).json({
+        status: 406,
         error: 'Account number not provided',
```
Comment (kampkelly) [inline server/controllers/accounts.controller.js]: Please change from 406.

File: server/tests/userSignUpTest.js [hunk 14]
```diff
@@ -1,8 +1,8 @@
+/* eslint-disable import/no-extraneous-dependencies */
 import { expect } from 'chai';
 import supertest from 'supertest';
 // import users from '../server/db/users.db';
 import {
```
Comment (kampkelly) [inline server/tests/userSignUpTest.js]: Please remove this comment.

File: server/utils/helpers.js [hunk 15]
```diff
@@ -1,14 +1,30 @@
...
 export const setAuthToken = (req, token) => {
   if (token) {
     req.headers['x-access-token'] = token;
   } else {
     req.headers['x-access-token'] = '';
   }
 };
+
+// Hash password
+export function hash(password) {
```
Comment (kampkelly) [inline server/utils/helpers.js]: You can remove the comments in this file.

File: server/validation/bankAccount.js [hunk 16]
```diff
@@ -32,14 +31,6 @@ const validateCreateBankAccountInput = (data) => {
     error = 'Email cannot be blank';
```
Comment (kampkelly) [inline server/validation/bankAccount.js]: You can remove this commented line.

**Pull request merged**